Tests/NIOTests/EventLoopFutureTest.swift

```diff
@@ -32,6 +32,166 @@ class EventLoopFutureTest : XCTestCase {
         let f = EventLoopFuture<Void>(eventLoop: eventLoop, error: EventLoopFutureTestError.example, file: #file, line: #line)
         XCTAssertTrue(f.isFulfilled)
     }
+    
+    func testFoldWithSuccessAndAllSuccesses() throws {
```

Comment:
could we have a test case that `fold`s `EventLoopFuture`s from different actual event loops (not `EmbeddedEventLoop`s)?

Reply:
Yes sure let me add one